src/routes/marketplace/[marketplaceDeckId]/+page.server.ts

```diff
@@ -0,0 +1,38 @@
+import { readPath, writePath } from "$lib/server/firebaseUtils";
+import { error, redirect } from "@sveltejs/kit";
+import type { PageServerLoad, Actions } from "./$types";
+import { getStripeCustomerWithSubscriptions, isSubscribedToBlendPro } from "$lib/server/subscriptionUtils";
+import { user } from "$lib/firebase";
+import { get } from "svelte/store";
+
+export const load = (async ({params: { marketplaceDeckId } }) => {
+  const marketplaceDeck = (await readPath(`/decks/marketplace/${marketplaceDeckId}`)) ?? {};
+  if (!marketplaceDeck)
+    throw error(404);
+  return {
+    deckName: marketplaceDeck.deck.name
+  }
```

Comment:
I believe this will be faster if you just fetch exactly what is needed from Firebase
```suggestion
  const deckName = (await readPath(`/decks/marketplace/${marketplaceDeckId}/deck/name`)) ?? {};
  if (!deckName )
    throw error(404);
  return {
    deckName
  }
```